content/docs/iac/using-pulumi/continuous-delivery/argocd.md

```diff
@@ -0,0 +1,398 @@
+
+```bash
+kubectl create namespace argocd
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+```
+
+Alternatively, use the Helm chart for more configuration options:
+
+```bash
+helm repo add argo https://argoproj.github.io/argo-helm
+helm install argocd argo/argo-cd -n argocd --create-namespace
+```
+
+### 2. Create a Pulumi Stack manifest
+
+Create a Kubernetes manifest file that defines your Pulumi Stack. This example creates a simple web application:
+
+```yaml
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pulumi-stack-sa
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: pulumi-stack-sa:system:auth-delegator
+  annotations:
+    argocd.argoproj.io/sync-wave: "1"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: pulumi-stack-sa
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: pulumi-stack-sa:cluster-admin
+  annotations:
+    argocd.argoproj.io/sync-wave: "1"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: pulumi-stack-sa
+  namespace: default
+---
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: webapp-dev
+  namespace: default
+  annotations:
+    argocd.argoproj.io/sync-wave: "2"
+    link.argocd.argoproj.io/external-link: https://app.pulumi.com/myorg/webapp/dev
+spec:
+  serviceAccountName: pulumi-stack-sa
+  stack: myorg/webapp/dev
+  projectRepo: "https://github.com/myorg/pulumi-webapp.git"
+  branch: main
+  refresh: true
+  continueResyncOnCommitMatch: true
+  resyncFrequencySeconds: 300
+  destroyOnFinalize: true
+  envRefs:
+    PULUMI_ACCESS_TOKEN:
+      type: Secret
+      secret:
+        name: pulumi-access-token
+        key: token
+  config:
+    webapp:environment: development
+    webapp:replicas: "2"
+```
+
+### 3. Create an ArgoCD Application
+
+Create an ArgoCD Application that monitors your Git repository:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: webapp-dev
+  namespace: argocd
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io/background
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/myorg/webapp-infrastructure.git
+    targetRevision: main
+    path: manifests/dev
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: default
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+    syncOptions:
+    - CreateNamespace=true
+```
+
+### 4. Configure secrets
+
+Create a Kubernetes Secret containing your Pulumi access token:
+
+```bash
+kubectl create secret generic pulumi-access-token \
+  --from-literal=token=$PULUMI_ACCESS_TOKEN \
+  -n default
+```
+
+## Advanced patterns
+
+### Multi-cluster deployments
```

Comment:
Please do this with `ApplicatonSets` and using a generator for this. Thats the preferred way of Fleet management.

Reply:
Will follow up post-merge with an update.